src/link/inline.js

```diff
@@ -1,60 +1,154 @@
-import { Popover, TextControl, Button } from '@wordpress/components';
+import {
+	Popover,
+	TextControl,
+	Button,
+	KeyboardShortcuts,
+} from '@wordpress/components';
 import { getTextContent, slice } from '@wordpress/rich-text';
 import { useState, useEffect } from '@wordpress/element';
 import { __ } from '@wordpress/i18n';
 import { getSiteLanguage } from './utils';
+import { prefixSearch, fulltextSearch } from './api';
 
 export const InlineEditUI = ( {
 	anchorRef,
 	onClose,
 	onApply,
-	onRemove,
 	value,
 	activeAttributes,
 } ) => {
 	const [ title, setTitle ] = useState( activeAttributes.title );
 	const [ lang, setLang ] = useState( activeAttributes.lang );
+	const [ searchList, setSearchList ] = useState( [] );
+	const [ hoveredIndex, setHoverIndex ] = useState( -1 );
+	const onItemHovered = ( index ) => {
+		if ( ! searchList.length ) return;
+		const className = 'wikipediapreview-edit-inline-list-item';
+
+		if ( hoveredIndex >= 0 ) {
+			document
+				.querySelectorAll( `.${ className }` )
```

Comment:
It's not really the react way to access the DOM like this.

You're already keeping the hoveredIndex in a state so the code that renders the list items can add the `hovered` class to the right item. There's a bunch or re-rendering anyway.